libs/base/Data/Fin.idr

```diff
@@ -15,47 +17,46 @@ data Fin : (n : Nat) -> Type where
     FS : Fin k -> Fin (S k)
 
 export
-implementation Uninhabited (Fin Z) where
+Uninhabited (Fin Z) where
   uninhabited FZ impossible
   uninhabited (FS f) impossible
 
 export
-FSInjective : (m : Fin k) -> (n : Fin k) -> FS m = FS n -> m = n
-FSInjective left _ Refl = Refl
+Uninhabited (FZ = FS k) where
+  uninhabited Refl impossible
 
 export
-implementation Eq (Fin n) where
-    (==) FZ FZ = True
-    (==) (FS k) (FS k') = k == k'
-    (==) _ _ = False
+Uninhabited (FS k = FZ) where
+  uninhabited Refl impossible
 
-||| There are no elements of `Fin Z`
 export
-FinZAbsurd : Fin Z -> Void
-FinZAbsurd FZ impossible
+FSInjective : FS m = FS n -> m = n
+FSInjective Refl = Refl
 
```

Comment:
Given the naming convention we seem to be adopting for `(succ/left/right/just)Injective`,
we should probably call this one `fsInjective`.